src/Kobo/SystemDialog.cpp

```diff
@@ -87,9 +91,25 @@ SystemWidget::Prepare(ContainerWindow &parent, const PixelRect &rc) noexcept
 inline void
 SystemWidget::SwitchOTGMode()
 {
-#ifdef KOBO
-  if (DetectKoboModel() != KoboModel::CLARA_HD) return SwitchKernel();
-#endif
+  if (DetectKoboModel() == KoboModel::CLARA_HD) {
+    mount("none", "/sys/kernel/debug", "debugfs", 0, "");
+    if (IsKoboOTGHostMode()) {
+      File::WriteExisting(Path("/sys/kernel/debug/ci_hdrc.0/role"), "gadget");
+      File::Delete(Path("/mnt/onboard/XCSoarData/kobo/OTG_Host_Active"));
+      switch_otg_mode->SetCaption("Enable USB-OTG");
```

Comment:
There is on error handling; we shouldn't change the state of the UI just by guessing that this has worked.
First, there should be error handling. If an error occurs, the error should be reported to the user.
Second, the state of the UI should always reflect the state of the system. i.e. even if the operation seemed to have succeeded, check the system state again (call `IsKoboOTGHostMode()` again).

Reply:
Will change shortly...

Reply:
Added error handling on the mode change.  Note that optimisation in SwitchOTGMode() cached the result of IsKoboOTGHostMode() and caused an incorrect failure indication (as host mode changes in the function).  I used the __attribute__ to disable optimisation for SwitchOTGMode() and it works.

Reply:
This is a bad solution; first, don't use `__attribute__`; it is a proprietary GCC extension. Even if we currently only use GCC, we should avoid using those extensions.
Second, disabling compiler optimization is a fragile kludge which steps away from really fixing the cause. This is caused by `IsKoboCustomKernel()` having a `const` attribute, which is wrong since you added code to the function which considers mutable runtime state (i.e. your change was buggy). Therefore, changing `const` to `pure` will fix the problem properly.

Reply:
Also, please don't hard-code `/mnt/onboard/XCSoarData`; you should instead use the `LocalPath.hpp` library. Even if it's unlikely that this path ever changes.

Reply:
I just followed the lead of KoboExecNickel() in src/Kobo/System.cpp.  I assumed the mkdirs was because XCSoarData was not guaranteed to be there if XCSoar had not yet been started.  I am changing it to use the #defines but should we also change System.cpp in a separate PR?